### PR TITLE
BUG: Throw ValueError when the format kwarg is passed to the HDFStore…

### LINKED ISSUE #13291: format ignored in HDFStore.open
```
new_store = pd.HDFStore('data/test.h5', format='table')
new_store['df'] = pd.DataFrame(np.random.rand(5,1))
new_store.put('df2', pd.DataFrame(np.random.rand(5,1)), format='table')
print('df stored as table: %s' % new_store.get_storer('df').is_table)
print('df2 stored as table: %s' % new_store.get_storer('df2').is_table)
```

df stored as table: False
df2 stored as table: True
## INSTALLED VERSIONS

commit: None
python: 2.7.11.final.0
python-bits: 64
OS: Linux
OS-release: 3.16.0-30-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.18.0

### PR DESCRIPTION
**Pull request by CianciuStyles**
… constructor

- [x] closes #13291
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.24.0.rst [hunk 1]
```diff
@@ -1492,6 +1492,7 @@ Notice how we now instead output ``np.nan`` itself instead of a stringified form
 - Bug in :func:`DataFrame.to_string()` that broke column alignment when ``index=False`` and width of first column's values is greater than the width of first column's header (:issue:`16839`, :issue:`13032`)
 - Bug in :func:`DataFrame.to_string()` that caused representations of :class:`DataFrame` to not take up the whole window (:issue:`22984`)
 - Bug in :func:`DataFrame.to_csv` where a single level MultiIndex incorrectly wrote a tuple. Now just the value of the index is written (:issue:`19589`).
+- :func:`HDFStore` will throw ``ValueError`` when the ``format`` kwarg is passed to the constructor (:issue:`13291`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.rst]: throw -> raise

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
looks ok, ping when pushed / green.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. ping on green.

**Comment by CianciuStyles** [comment]
Hello Jeff, it looks like everything is green now.

**Comment by jreback** [comment]
thans @CianciuStyles

**Pull request merged**